enclave_wrangler/dataset_download_new_api.py

```diff
@@ -12,67 +13,116 @@
 from enclave_wrangler.config import config
 # from enclave_wrangler.utils import log_debug_info
 
+
 HEADERS = {
     "authorization": f"Bearer {config['PALANTIR_ENCLAVE_AUTHENTICATION_BEARER_TOKEN']}",
-    #'content-type': 'application/json'
+    # 'content-type': 'application/json'
 }
 DEBUG = False
 # TARGET_CSV_DIR='data/datasets/'
 
 
-@typechecked
-def objTypes() -> [{}]:
-    """API documentation at
-    https://www.palantir.com/docs/foundry/api/ontology-resources/object-types/list-object-types/
-    """
-
-    ontologyRid = config['ONTOLOGY_RID']
-    api_path = f'/api/v1/ontologies/{ontologyRid}/objectTypes'
-    url = f'https://{config["HOSTNAME"]}{api_path}'
-
-    response = requests.get(url, headers=HEADERS,)
-    response_json = response.json()
-    # print(response_json)
-    # types = pd.DataFrame(data=response_json)
-    return response_json['data']
-
-
-def run() -> None:
+class EnclaveClient:
```

Comment:
I generally use OO whenever I feel like it's necessary. It seems pretty common for HTTP clients to be implemented in this way, and it has been semi-helpful for me to do this refactor.